transport/src/main/java/io/netty/channel/nio/NioEventLoop.java

```diff
@@ -377,6 +343,48 @@ protected void run() {
         }
     }
 
+    protected boolean doConditionalSelect(boolean oldWakenUp) throws Exception {
+        if (hasTasks()) {
+            selectNow();
+        } else {
+            select(oldWakenUp);
+
+            // 'wakenUp.compareAndSet(false, true)' is always evaluated
+            // before calling 'selector.wakeup()' to reduce the wake-up
+            // overhead. (Selector.wakeup() is an expensive operation.)
+            //
+            // However, there is a race condition in this approach.
+            // The race condition is triggered when 'wakenUp' is set to
+            // true too early.
+            //
+            // 'wakenUp' is set to true too early if:
+            // 1) Selector is waken up between 'wakenUp.set(false)' and
+            //    'selector.select(...)'. (BAD)
+            // 2) Selector is waken up between 'selector.select(...)' and
+            //    'if (wakenUp.get()) { ... }'. (OK)
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) This block of commented-out lines of code should be removed. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3ACommentedOutCodeLine)